`EoSyntax` fails when debug mode is enabled

I've enabled debug mode (`-X`) using the following command in the `fibonacci` integration test:

```
invoker.goals = clean test -X
```

and got the following output:

```
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 2 out of bounds for length 2
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at com.jcabi.log.PreFormatter.process (PreFormatter.java:124)
    at com.jcabi.log.PreFormatter.<init> (PreFormatter.java:76)
    at com.jcabi.log.Logger.format (Logger.java:121)
    at com.jcabi.log.Logger.debugForced (Logger.java:202)
    at com.jcabi.log.Logger.debug (Logger.java:187)
    at org.eolang.parser.EoSyntax.parsed (EoSyntax.java:205)
    at org.eolang.maven.MjParse.parsed (MjParse.java:145)
    at org.eolang.maven.MjParse.lambda$parsed$0 (MjParse.java:105)
    at org.eolang.maven.FpDefault.lambda$new$1 (FpDefault.java:72)
    at org.cactoos.func.UncheckedFunc.lambda$apply$0 (UncheckedFunc.java:37)
    at org.cactoos.scalar.Checked.value (Checked.java:56)
    at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
    at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
    at org.cactoos.func.UncheckedFunc.apply (UncheckedFunc.java:38)
    at org.eolang.maven.FpGenerated.apply (FpGenerated.java:33)
    at org.eolang.maven.FpFork.lambda$new$1 (FpFork.java:44)
    at org.eolang.maven.FpEnvelope.apply (FpEnvelope.java:31)
    at org.eolang.maven.FpEnvelope.apply (FpEnvelope.java:31)
    at org.eolang.maven.FpExistedSource.lambda$new$0 (FpExistedSource.java:25)
    at org.eolang.maven.FpEnvelope.apply (FpEnvelope.java:31)
    at org.eolang.maven.FpEnvelope.apply (FpEnvelope.java:31)
    at org.eolang.maven.MjParse.parsed (MjParse.java:114)
    at org.eolang.maven.Threaded.lambda$total$1 (Threaded.java:76)
    at org.cactoos.scalar.CallableEnvelope.call (CallableEnvelope.java:34)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)
```

**Expected behaviour:** the integration test runs successfully

--- a/eo-parser/pom.xml
+++ b/eo-parser/pom.xml
@@ -109,7 +109,12 @@
     <dependency>
       <groupId>log4j</groupId>
       <artifactId>log4j</artifactId>
-      <!-- version from parent POM -->
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>ch.qos.reload4j</groupId>
+      <artifactId>reload4j</artifactId>
+      <version>1.2.22</version>
       <scope>test</scope>
     </dependency>
     <dependency>

--- a/eo-parser/src/main/java/org/eolang/parser/EoSyntax.java
+++ b/eo-parser/src/main/java/org/eolang/parser/EoSyntax.java
@@ -203,7 +203,7 @@ public final class EoSyntax implements Syntax {
             );
         } else {
             Logger.debug(
-                this, "The %s program of %d EO lines compiled with %d error(s)",
+                this, "The program of %d EO lines compiled with %d error(s)",
                 lines.size(), errors
             );
         }

--- a/eo-parser/src/test/java/org/eolang/parser/EoSyntaxTest.java
+++ b/eo-parser/src/test/java/org/eolang/parser/EoSyntaxTest.java
@@ -18,6 +18,7 @@ import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import org.apache.commons.text.StringEscapeUtils;
+import org.apache.log4j.Level;
 import org.cactoos.io.InputOf;
 import org.cactoos.io.ResourceOf;
 import org.cactoos.iterable.Mapped;
@@ -31,10 +32,14 @@ import org.eolang.xax.Xtory;
 import org.eolang.xax.XtoryMatcher;
 import org.hamcrest.MatcherAssert;
 import org.hamcrest.Matchers;
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.CsvSource;
@@ -48,7 +53,25 @@ import org.xml.sax.SAXParseException;
  * @since 0.1
  */
 @SuppressWarnings("PMD.TooManyMethods")
+@Execution(ExecutionMode.SAME_THREAD)
 final class EoSyntaxTest {
+
+    /**
+     * Log level before test.
+     */
+    private Level before;
+
+    @BeforeEach
+    void setUp() {
+        final org.apache.log4j.Logger logger = org.apache.log4j.Logger.getLogger(EoSyntax.class);
+        this.before = logger.getLevel();
+    }
+
+    @AfterEach
+    void tearDown() {
+        final org.apache.log4j.Logger logger = org.apache.log4j.Logger.getLogger(EoSyntax.class);
+        logger.setLevel(this.before);
+    }
 
     @Test
     void parsesSimpleCode() throws Exception {
@@ -68,6 +91,23 @@ final class EoSyntaxTest {
                 "/object/metas/meta[head='meta2']",
                 "/object/o[@name='fibo']"
             )
+        );
+    }
+
+    @Test
+    void parsesSimpleCodeWithDebugMode() {
+        final org.apache.log4j.Logger logger = org.apache.log4j.Logger.getLogger(EoSyntax.class);
+        logger.setLevel(Level.DEBUG);
+        final EoSyntax syntax = new EoSyntax(
+            String.join(
+                "\n",
+                "# No comments.",
+                "[] > x-н, 1\n"
+            )
+        );
+        Assertions.assertDoesNotThrow(
+            syntax::parsed,
+            "EO syntax should not fail in debug mode when program has errors"
         );
     }
 

